Make it easier to start "cross-test" helper apps in Darwin tests.

bzbarsky-apple opened the pull request:
We shouldn't require a testcase instance to start such an app, since it's associated with the test suite, not a specific test.

bzbarsky-apple committed:
Make it easier to start "cross-test" helper apps in Darwin tests.

We shouldn't require a testcase instance to start such an app, since it's
associated with the test suite, not a specific test.

ksperling-apple commented on src/darwin/Framework/CHIPTests/TestHelpers/MTRTestServerAppRunner.h:
```diff
@@ -48,8 +48,10 @@ NS_ASSUME_NONNULL_BEGIN
+ * The passed-in testsuite must be a subclass of MTRTestCase.
  */
-- (instancetype)initCrossTestWithAppName:(NSString *)name arguments:(NSArray<NSString *> *)arguments payload:(NSString *)payload testcase:(MTRTestCase *)testcase;
+- (instancetype)initCrossTestWithAppName:(NSString *)name arguments:(NSArray<NSString *> *)arguments payload:(NSString *)payload testsuite:(Class)testsuite;
```
Instead of passing around the test case class and having different initializers for suite vs test scope, have you considered making this functionality available as a category on MTRTestCase?

bzbarsky-apple replied:
Hmm, without involving an apprunner object at all?  Let me think about it.

bzbarsky-apple replied:
Yeah, this is a good idea; I'll do that.

bzbarsky-apple committed:
Address review comment.

Switch to just having app-starting functionality as a category on MTRTestCase.
This is much easier to use.

ksperling-apple commented on src/darwin/Framework/CHIPTests/TestHelpers/MTRTestServerAppRunner.h:
```diff
@@ -16,21 +16,14 @@
- * with the binary being out/debug/${APPNAME}-app/chip-${APPNAME}-app.
- */
-@interface MTRTestServerAppRunner : NSObject
+@interface MTRTestCase (ServerAppRunner)
```
Should probably rename this file to `MTRTestCase+ServerAppRunner.h` now

bzbarsky-apple replied:
Yep, done.

ksperling-apple reviewed (approved):
Looks great!

bzbarsky-apple committed:
Rename MTRTestServerAppRunner to MTRTestCase+ServerAppRunner.
